drivers/vulkan/rendering_device_driver_vulkan.h

```diff
@@ -444,7 +445,15 @@ class RenderingDeviceDriverVulkan : public RenderingDeviceDriver {
 public:
 	virtual String shader_get_binary_cache_key() override final;
 	virtual Vector<uint8_t> shader_compile_binary_from_spirv(VectorView<ShaderStageSPIRVData> p_spirv, const String &p_shader_name) override final;
-	virtual ShaderID shader_create_from_bytecode(const Vector<uint8_t> &p_shader_binary, ShaderDescription &r_shader_desc, String &r_name) override final;
+	/** Creates a Pipeline State Object (PSO) out of the shader and all the input data it needs.
```

Comment:
It's good to have some documentation, but Godot is not following this convention. I don't know what should we do.

Reply:
There are cases where Godot follows doxygen documentation, such as:

- gdextension_interface.h
- `split_by_plane` in godot/core/math/face3.h

I followed those same conventions.

I know it's rare and Godot doesn't have code comments (which is very bad) and there are no standards set; but the `r_immutableSamplers` parameter was simply important to clarify the gotchas (that caller promises not to bind a different sampler because that simply won't work).

Reply:
Yeah, makes sense. Maybe this eventually becomes the standard to follow. In the meantime, there's no real reson to discuss it. I just stroke my eye.